include/license.php

```diff
@@ -84,10 +84,12 @@ public function __construct( $file, $item_name, $version, $author, $api_url = nu
 		$this->author  = $author;
 		$this->api_url = empty( $api_url ) ? $this->api_url : $api_url;
 
-		$licenses = get_option( 'polylang_licenses' );
-		$this->license_key = empty( $licenses[ $this->id ]['key'] ) ? '' : $licenses[ $this->id ]['key'];
-		if ( ! empty( $licenses[ $this->id ]['data'] ) ) {
-			$this->license_data = $licenses[ $this->id ]['data'];
+		$licenses          = (array) get_option( 'polylang_licenses', array() );
+		$license           = isset( $licenses[ $this->id ] ) && is_array( $licenses[ $this->id ] ) ? $licenses[ $this->id ] : array();
+		$this->license_key = ! empty( $license['key'] ) && is_string( $license['key'] ) ? $license['key'] : '';
+
+		if ( ! empty( $license['data'] ) && $license['data'] instanceof stdClass ) {
+			$this->license_data = $license['data'];
```

Comment:
Wouldn't this work 
```php
$this->license_key = (string) $license['key'];
$this->license_data = (object) $license['data'];
```

Reply:
`key` and `data` may not be set.

Reply:
Sure. However, I find:
```php
		$this->license_key = ! empty( $license['key'] ) ? (string) $license['key'] : '';
		
		if ( ! empty( $license['data'] ) ) {
			$this->license_data = (object) $license['data'];
		}
```
more elegant than type check. It's however not strictly equivalent to what you wrote.
As you prefer.